llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -4613,15 +4613,41 @@ static bool isSubRegOf(const SIRegisterInfo &TRI,
          SubReg.getReg() == SuperVec.getReg();
 }
 
+// Verify the illgal copy from VGPR to SGPR for generic opcode COPY
+bool SIInstrInfo::verifyCopy(const MachineInstr &MI,
+                             const MachineRegisterInfo &MRI,
+                             StringRef &ErrInfo) const {
+  const MachineOperand &Dst = MI.getOperand(0);
+  const MachineOperand &Src = MI.getOperand(1);
+
+  if (Dst.isReg() && Src.isReg()) {
+    Register DstReg = Dst.getReg();
+    Register SrcReg = Src.getReg();
+    // This is a check for copy from an VGPR to SGPR
+    if (RI.isVGPR(MRI, SrcReg) && RI.isSGPRReg(MRI, DstReg)) {
```

Comment:
Does this also diagnose AGPR to SGPR copies?

Reply:
Also, this only validates in one direction. shouldn't we check the other way as well?
SGPR -> VectorClass
VectorClass -> SGPR

Reply:
No! COPY from SGPR to VGPR is fine.

Reply:
Added the illegal copy check from AGPR to SGPR.